engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+INCLUDE "constants.asm"
+
+SECTION "engine/movie/opening_cutscene.asm", ROMX
+
+OpeningCutscene::
+	call .Init
+.loop
+	call .PlayFrame
+	jr nc, .loop
+	ret
+
+.Init:
+	callba InitEffectObject
+	xor a
+	ld [wIntroJumptableIndex], a
+	ldh [hBGMapMode], a
+	ret
+
+.PlayFrame:
+	ld hl, hJoypadDown
+	ld a, [hl]
+	and %00001111
+	jr nz, .Finish
+
+; check done flag
+	ld a, [wIntroJumptableIndex]
+	bit 7, a
+	jr nz, .Finish
+
+	callba EffectObjectJumpNoDelay
+	call IntroSceneJumper
+	call DelayFrame
+	and a
+	ret
+
+.Finish:
+	callab InitEffectObject
+	call ClearSprites
+	call DelayFrame
+	xor a
+	ldh [hSCX], a
+	ldh [hSCY], a
+	ldh [hLCDCPointer], a
+	ldh [hLYOverrideStart], a
+	ldh [hLYOverrideEnd], a
+	ld a, %11100100
+	ldh [rBGP], a
+	ldh [rOBP0], a
+	scf
+	ret
+
+IntroSceneJumper:
+	jumptable .scenes, wIntroJumptableIndex
+.scenes
+	dw IntroScene1
+	dw IntroScene2
+	dw IntroScene3
+	dw IntroScene4
+	dw IntroScene5
+	dw IntroScene6
+	dw IntroScene7
+	dw IntroScene8
+	dw IntroScene9
+	dw IntroScene10
+	dw IntroScene11
+	dw IntroScene12
+	dw IntroScene13
+	dw IntroScene14
+	dw IntroScene15
+	dw IntroScene16
+	dw IntroScene17
+
+IntroScene1:	; 43b8
+; Set up water cutscene
+	ld hl, wIntroJumptableIndex
+	inc [hl]
+	call DisableLCD
+
+	ld b, SGB_GS_INTRO
+	ld c, 0
+	call GetSGBLayout
+
+	callab InitEffectObject
+
+	call Intro_ResetLYOverrides
+
+	ld hl, vChars2
+	ld de, IntroUnderwaterGFX
+	call Intro_Copy128Tiles
+
+; Load water metatiles
+	ld a, LOW(Intro_WaterMeta)
+	ld [wIntroTilesPointer + 0], a
+	ld a, HIGH(Intro_WaterMeta)
+	ld [wIntroTilesPointer + 1], a
+
+; Set destination BG map pointer
+	ld hl, vBGMap0
+	ld a, l
+	ld [wIntroBGMapPointer + 0], a
+	ld a, h
+	ld [wIntroBGMapPointer + 1], a
+
+; Load water tilemap
+	ld de, Intro_WaterTilemap + 15 tiles
```

Comment:
`Intro_WaterTilemap + 15 tiles`? Are the first $F0 bytes of water_tilemap.bin really part of the tilemap?

Reply:
(Also `tiles` is used for offsetting 2bpp graphics anyway, where one tile is 16 bytes; this would be `Intro_WaterTilemap + $f0`)